File: app/src/main/java/com/cere/csplayer/view/CustomViews.java
package com.cere.csplayer.view;

import android.graphics.Bitmap;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.graphics.Path;

import androidx.annotation.ColorInt;

/**
 * Created by CheRevir on 2019/10/26
 */
public class CustomViews {
    public static Bitmap getOldToNewBitmap(Bitmap bitmap, @ColorInt int oldColor, @ColorInt int newColor) {
        Bitmap newBitmap = Bitmap.createBitmap(bitmap.getWidth(), bitmap.getHeight(), Bitmap.Config.ARGB_4444);
        for (int x = 0; x < bitmap.getWidth(); x++) {
            for (int y = 0; y < bitmap.getHeight(); y++) {
                if (bitmap.getPixel(x, y) == oldColor) {
                    newBitmap.setPixel(x, y, newColor);
                } else {
                    newBitmap.setPixel(x, y, bitmap.getPixel(x, y));
                }
            }
        }
        bitmap.recycle();
        return newBitmap;
    }

    public static Bitmap getPrev(@ColorInt int color) {
        Bitmap bitmap = Bitmap.createBitmap(100, 100, Bitmap.Config.ARGB_8888);
        Paint paint = new Paint();
        paint.setAntiAlias(true);
        paint.setColor(color);
        paint.setStyle(Paint.Style.FILL);
        Canvas canvas = new Canvas(bitmap);
        Path path = new Path();
        path.moveTo(15, 50);
        path.lineTo(50, 25);
        path.lineTo(50, 75);
        path.moveTo(50, 50);
        path.lineTo(85, 25);
        path.lineTo(85, 75);
        path.close();
        canvas.drawPath(path, paint);
        return bitmap;
    }

    public static Bitmap getNext(@ColorInt int color) {
        Bitmap bitmap = Bitmap.createBitmap(100, 100, Bitmap.Config.ARGB_8888);
        Paint paint = new Paint();
        paint.setAntiAlias(true);
        paint.setColor(color);
        paint.setStyle(Paint.Style.FILL);
        Canvas canvas = new Canvas(bitmap);
        Path path = new Path();
        path.moveTo(15, 25);
        path.lineTo(15, 75);
        path.lineTo(50, 50);
        path.moveTo(50, 25);
        path.lineTo(50, 75);
        path.lineTo(85, 50);
        path.close();
        canvas.drawPath(path, paint);
        return bitmap;
    }

    public static Bitmap getPause(@ColorInt int color) {
        Bitmap bitmap = Bitmap.createBitmap(100, 100, Bitmap.Config.ARGB_8888);
        Paint paint = new Paint();
        paint.setAntiAlias(true);
        paint.setColor(color);
        paint.setStyle(Paint.Style.FILL);
        Canvas canvas = new Canvas(bitmap);
        Path path = new Path();
        path.moveTo(15, 10);
        path.lineTo(15, 90);
        path.lineTo(85, 50);
        path.close();
        canvas.drawPath(path, paint);
        return bitmap;
    }

    public static Bitmap getPlay(@ColorInt int color) {
        Bitmap bitmap = Bitmap.createBitmap(100, 100, Bitmap.Config.ARGB_8888);
        Paint paint = new Paint();
        paint.setAntiAlias(true);
        paint.setColor(color);
        paint.setStyle(Paint.Style.FILL);
        Canvas canvas = new Canvas(bitmap);
        canvas.drawRect(15, 10, 40, 90, paint);
        canvas.drawRect(60, 10, 85, 90, paint);
        return bitmap;
    }

    public static Bitmap getClose(@ColorInt int color) {
        Bitmap bitmap = Bitmap.createBitmap(100, 100, Bitmap.Config.ARGB_8888);
        Paint paint = new Paint();
        paint.setAntiAlias(true);
        paint.setColor(color);
        paint.setStrokeWidth(10);
        paint.setStyle(Paint.Style.FILL);
        Canvas canvas = new Canvas(bitmap);
        canvas.drawLine(15, 10, 85, 90, paint);
        canvas.drawLine(15, 90, 85, 10, paint);
        return bitmap;
    }
}

File: app/src/main/java/com/cere/csplayer/view/CustomViews.kt
package com.cere.csplayer.view

import android.graphics.Bitmap
import android.graphics.Canvas
import android.graphics.Paint
import android.graphics.Path
import androidx.annotation.ColorInt

/**
 * Created by CheRevir on 2019/10/26
 */
object CustomViews {
    @JvmStatic
    fun getOldToNewBitmap(bitmap: Bitmap, @ColorInt oldColor: Int, @ColorInt newColor: Int): Bitmap {
        val newBitmap = Bitmap.createBitmap(bitmap.width, bitmap.height, Bitmap.Config.ARGB_8888)
        for (x in 0 until bitmap.width) {
            for (y in 0 until bitmap.height) {
                if (bitmap.getPixel(x, y) == oldColor) {
                    newBitmap.setPixel(x, y, newColor)
                } else {
                    newBitmap.setPixel(x, y, bitmap.getPixel(x, y))
                }
            }
        }
        bitmap.recycle()
        return newBitmap
    }

    @JvmStatic
    fun getPrev(@ColorInt color: Int): Bitmap {
        val bitmap = Bitmap.createBitmap(100, 100, Bitmap.Config.ARGB_8888)
        val paint = Paint()
        paint.isAntiAlias = true
        paint.color = color
        paint.style = Paint.Style.FILL
        val canvas = Canvas(bitmap)
        val path = Path()
        path.moveTo(15f, 50f)
        path.lineTo(50f, 25f)
        path.lineTo(50f, 75f)
        path.moveTo(50f, 50f)
        path.lineTo(85f, 25f)
        path.lineTo(85f, 75f)
        path.close()
        canvas.drawPath(path, paint)
        return bitmap
    }

    @JvmStatic
    fun getNext(@ColorInt color: Int): Bitmap {
        val bitmap = Bitmap.createBitmap(100, 100, Bitmap.Config.ARGB_8888)
        val paint = Paint()
        paint.isAntiAlias = true
        paint.color = color
        paint.style = Paint.Style.FILL
        val canvas = Canvas(bitmap)
        val path = Path()
        path.moveTo(15f, 25f)
        path.lineTo(15f, 75f)
        path.lineTo(50f, 50f)
        path.moveTo(50f, 25f)
        path.lineTo(50f, 75f)
        path.lineTo(85f, 50f)
        path.close()
        canvas.drawPath(path, paint)
        return bitmap
    }

    @JvmStatic
    fun getPause(@ColorInt color: Int): Bitmap {
        val bitmap = Bitmap.createBitmap(100, 100, Bitmap.Config.ARGB_8888)
        val paint = Paint()
        paint.isAntiAlias = true
        paint.color = color
        paint.style = Paint.Style.FILL
        val canvas = Canvas(bitmap)
        val path = Path()
        path.moveTo(15f, 10f)
        path.lineTo(15f, 90f)
        path.lineTo(85f, 50f)
        path.close()
        canvas.drawPath(path, paint)
        return bitmap
    }

    @JvmStatic
    fun getPlay(@ColorInt color: Int): Bitmap {
        val bitmap = Bitmap.createBitmap(100, 100, Bitmap.Config.ARGB_8888)
        val paint = Paint()
        paint.isAntiAlias = true
        paint.color = color
        paint.style = Paint.Style.FILL
        val canvas = Canvas(bitmap)
        canvas.drawRect(15f, 10f, 40f, 90f, paint)
        canvas.drawRect(60f, 10f, 85f, 90f, paint)
        return bitmap
    }

    @JvmStatic
    fun getClose(@ColorInt color: Int): Bitmap {
        val bitmap = Bitmap.createBitmap(100, 100, Bitmap.Config.ARGB_8888)
        val paint = Paint()
        paint.isAntiAlias = true
        paint.color = color
        paint.strokeWidth = 10f
        paint.style = Paint.Style.FILL
        val canvas = Canvas(bitmap)
        canvas.drawLine(15f, 10f, 85f, 90f, paint)
        canvas.drawLine(15f, 90f, 85f, 10f, paint)
        return bitmap
    }
}